docs/tutorials/getting-started-with-flask.rst

```diff
@@ -0,0 +1,114 @@
+Getting started with Rockcraft for Flask
+----------------------------------------
+
+In this tutorial you will learn how to create a Flask application, build a rock
+with the Flask application and run it.
+
+Prerequisites
+-------------
+- snap enabled system (https://snapcraft.io)
+- LXD installed (https://linuxcontainers.org/lxd/getting-started-cli/)
+- Docker installed (https://snapcraft.io/docker)
+- a text editor
+
+Create Flask application
+------------------------
+
+Create a new directory and write the following into a text editor and save it as
+``requirements.txt``:
+
+.. literalinclude:: code/getting-started-with-flask/requirements.txt
+
+Install ``python3-venv`` on your host and create a virtual environment:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-venv]
+    :end-before: [docs:create-venv-end]
+    :dedent: 2
+
+In the same directory, write the following into a text editor and save it as
+``app.py``:
+
+.. literalinclude:: code/getting-started-with-flask/app.py
+    :language: python
+
+Run and test the flask application to verify everything is working:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:run-flask]
+    :end-before: [docs:run-flask-end]
+    :dedent: 2
+
+The ``curl`` command should print ``Hello, world!``.
+
+Install Rockcraft
+-----------------
+
+Install Rockcraft on your host:
+
+.. literalinclude:: code/hello-world/task.yaml
+    :language: bash
+    :start-after: [docs:install-rockcraft]
+    :end-before: [docs:install-rockcraft-end]
+    :dedent: 2
+
+Create Flask rock
+-----------------
+
+Use the ``flask-framework`` profile to create ``rockcraft.yaml``:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-rockcraft-yaml]
+    :end-before: [docs:create-rockcraft-yaml-end]
+    :dedent: 2
+
+Open ``rockcraft.yaml`` in a text editor and customise the ``name``,
+``summary``, ``description`` and ``license``. Ensure that ``platforms`` includes
+the architecture of your host. For example, if your host uses the ARM
+architecture, include ``arm64`` in ``platforms``. Pack the rock:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
```

Comment:
why the iptables command? that shouldn't happen with the `docker` snap. And if it does, this is one more remark that belongs to a troubleshooting note/section

Reply:
the fix is explained here: https://documentation.ubuntu.com/lxd/en/latest/howto/network_bridge_firewalld/#prevent-connectivity-issues-with-lxd-and-docker

I've also noticed that doing `lxd init` first and then install `snap install docker` might avoid the issue, so if you can reproduce that, I'd make sure that's the right order in the docs

Reply:
yeh I did the same an this wasn't needed, done

Reply:
I just tested this in a multipass VM. However, it is just a matter of restarting the VM, and the failure appears again...